src/main/java/com/launchacademy/reviews/services/ReviewService.java

```diff
@@ -25,6 +25,13 @@ public Review save(Review review){
     return reviewRepository.save(review);
   }
 
+  public void deleteAllByFilmId(Integer id) {
+    List<Review> listOfReviews = reviewRepository.findAllByFilmId(id);
+    for (Integer i = 0; i < listOfReviews.size(); i++) {
+      reviewRepository.delete(listOfReviews.get(i));
+    }
+  }

```

Comment:
While this will work, as your database gets bigger this will start to drastically slow down.  I would look to take advantage of `orphanRemoval` as a refactor.

https://thorben-janssen.com/avoid-cascadetype-delete-many-assocations/